From a12de954a87a16f56284b59f9c4d78f2854f142f Mon Sep 17 00:00:00 2001
From: Axel Lin <axel.lin@ingics.com>
Date: Sun, 14 Feb 2016 00:19:07 +0800
Subject: [PATCH 099/101] spi: bcm2835: Remove unnecessary workaround to call
 gpio_set_value
Git-commit: 763dedfcda5f6c9294c4a5d048f5261f442cc1e8
Patch-mainline: v4.6-rc1
References: FATE#319481

This should be fixed by commit 4c02cba18cc9
("pinctrl: bcm2835: Fix initial value for direction_output")

Signed-off-by: Axel Lin <axel.lin@ingics.com>
Acked-by: Martin Sperl <kernel@martin.sperl.org>
Signed-off-by: Mark Brown <broonie@kernel.org>
Signed-off-by: Alexander Graf <agraf@suse.de>

---
 drivers/spi/spi-bcm2835.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/drivers/spi/spi-bcm2835.c b/drivers/spi/spi-bcm2835.c
index cf04960..f35cc10 100644
--- a/drivers/spi/spi-bcm2835.c
+++ b/drivers/spi/spi-bcm2835.c
@@ -727,11 +727,6 @@ static int bcm2835_spi_setup(struct spi_device *spi)
 			spi->chip_select, spi->cs_gpio, err);
 		return err;
 	}
-	/* the implementation of pinctrl-bcm2835 currently does not
-	 * set the GPIO value when using gpio_direction_output
-	 * so we are setting it here explicitly
-	 */
-	gpio_set_value(spi->cs_gpio, (spi->mode & SPI_CS_HIGH) ? 0 : 1);
 
 	return 0;
 }
-- 
1.8.5.6

